test/compression_cases_test.go

```diff
@@ -0,0 +1,108 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/internal/stubserver"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+	"google.golang.org/grpc/status"
+)
+
+type mockCompressor struct {
+	grpc.Compressor
+}
+
+func newMockCompressor() grpc.Compressor {
+	return &mockCompressor{grpc.NewGZIPCompressor()}
+}
+
+func (c *mockCompressor) Type() string {
+	return "mock"
+}
+
+type mockDecompressor struct {
+	grpc.Decompressor
+}
+
+func newMockDecompressor() grpc.Decompressor {
+	return &mockDecompressor{grpc.NewGZIPDecompressor()}
+}
+
+func (d *mockDecompressor) Type() string {
+	return "mock"
+}
+
+func TestCompressionCases(t *testing.T) {
```

Comment:
After you move this to the other file, run the tests through a tester struct which does extra stuff like tracking leaked go routines and configuring the logger. So you should change this to a method on the `grpctest.Tester` struct
```go
func (s) TestCompressionCases(t *testing.T) {
```
The test name gets modified to `Test/CompressionCases` instead of `TestCompressionCases` if you want to run it using `go test -run`